lib/dotcom_web/live/subway_alerts.ex

```diff
@@ -0,0 +1,115 @@
+defmodule DotcomWeb.Live.SubwayAlerts do
+  @moduledoc """
+  The subway alerts/status page, showing current subway status, planned work, and
+  other upcoming and relevant alerts.
+  """
+
+  use DotcomWeb, :live_view
+
+  import Dotcom.Alerts, only: [service_impacting_alert?: 1]
+  import DotcomWeb.Components.PlannedDisruptions, only: [disruptions: 1]
+  import DotcomWeb.Components.SystemStatus.SubwayStatus, only: [alerts_subway_status: 1]
+
+  alias Alerts.InformedEntity
+  alias Alerts.Match
+  alias Dotcom.Alerts.Subway.Disruptions
+  alias Dotcom.SystemStatus
+
+  @alerts_repo Application.compile_env!(:dotcom, :repo_modules)[:alerts]
+  @routes_repo Application.compile_env!(:dotcom, :repo_modules)[:routes]
+
+  @date_time Application.compile_env!(:dotcom, :date_time_module)
+
+  embed_templates "layouts/*"
+
+  def mount(_params, _session, socket) do
+    {:ok,
+     socket
+     |> assign(
+       :meta_description,
+       "Live service alerts for all MBTA transportation modes, including subway, bus, Commuter Rail, and ferry. " <>
+         "Updates on delays, construction, elevator outages, and more."
+     )
+     |> assign(:breadcrumbs, [Breadcrumb.build("Alerts")])
+     |> assign(:date_time, @date_time.now())
+     |> assign_banner_alert()
+     |> assign_result(&alert_groups/0)
+     |> assign_result(&SystemStatus.subway_status/0)
+     |> assign_result(&Disruptions.future_disruptions/0)}
+  end
+
+  def handle_params(%{"alerts_timeframe" => _} = params, _uri, socket) do
+    new_path = socket |> live_path(__MODULE__, params |> Map.drop(["alerts_timeframe"]))
+    {:noreply, socket |> push_patch(to: new_path)}
+  end
+
+  def handle_params(_params, _uri, socket) do
+    {:noreply, socket}
+  end
+
+  defp assign_banner_alert(socket) do
+    case @alerts_repo.banner() do
+      nil -> socket
+      banner -> socket |> assign(:alert_banner, banner)
+    end
+  end
+
+  defp alert_groups() do
+    alerts = @alerts_repo.all(@date_time.now()) |> Enum.reject(&service_impacting_alert?/1)
+    non_banner_alerts = excluding_banner(@alerts_repo.banner(), alerts)
+
+    [0, 1]
+    |> @routes_repo.by_type()
+    |> with_alert_lists(non_banner_alerts)
+    |> drop_empty_groups()
+  end
+
+  defp with_alert_lists(routes, alerts) do
+    routes
+    |> Enum.map(fn route ->
+      entity = %InformedEntity{
+        route_type: route.type,
+        route: route.id
+      }
```

Comment:
suggestion: there's a function for that :D

```suggestion
      entity = InformedEntity.from_keywords(
         route_type: route.type,
         route: route.id
      )
```

Reply:
[Cool!](https://github.com/mbta/dotcom/pull/2663/commits/ec95e52b7bfd7dba68b15f0c3886eabb5262532a)